Integration test broken

### Apache Iceberg version

main (development)

### Please describe the bug 🐞

As of [current commit](https://github.com/apache/iceberg-python/commit/36b56eb7bbc4cff91e4b9bfdf71ca0a7b8bbee5c) 3/2/2024

Error log:
```
>           assert spark_partition_for_justification == expected_partition_record
E           assert Record[timestamp_field_hour=464611] == Record[timestamp_field_hour=464603]
E             Full diff:
E             - Record[timestamp_field_hour=464603]
E             ?                                 ^^
E             + Record[timestamp_field_hour=464611]
E             ?                                 ^^

tests/integration/test_partitioning_key.py:771: AssertionError
```

Reproduce:
```
git checkout main
git pull
make test-integration
```

Possibly related to #453

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1961,6 +1961,7 @@ def spark() -> SparkSession:
 
     spark = (
         SparkSession.builder.appName("PyIceberg integration test")
+        .config("spark.sql.session.timeZone", "UTC")
         .config("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
         .config("spark.sql.catalog.integration", "org.apache.iceberg.spark.SparkCatalog")
         .config("spark.sql.catalog.integration.catalog-impl", "org.apache.iceberg.rest.RESTCatalog")

